colibre/registration.py

```diff
@@ -24,333 +24,464 @@
-    setattr(self, "gas_H2_plus_He_mass", H2_mass_wHe)
-
-except AttributeError:
-    # We did not produce these quantities.
-    setattr(
-        self,
-        "gas_H2_mass",
-        unyt.unyt_array(
-            catalogue.masses.m_gas, name="$M{\\rm H_2}$ not found, showing $M_{\\rm g}$"
-        ),
-    )
-
-# Get neutral H masses and fractions
-try:
-    gas_mass = catalogue.masses.m_gas
-    H_frac = getattr(catalogue.element_mass_fractions, "element_0")
-
-    # Try CHIMES arrays
-    if hasattr(catalogue.species_fractions, "species_7"):
-        HI_frac = getattr(catalogue.species_fractions, "species_1")
-        H2_frac = getattr(catalogue.species_fractions, "species_7")
-    # If species_7 doesn't exist, switch to the (default) Table-cooling case
-    else:
-        HI_frac = getattr(catalogue.species_fractions, "species_0")
-        H2_frac = getattr(catalogue.species_fractions, "species_2")
-
-    HI_mass = gas_mass * H_frac * HI_frac
-    H2_mass = gas_mass * H_frac * H2_frac * 2
-    neutral_H_mass = HI_mass + H2_mass
-    neutral_H_mass.name = "$M_{\\rm HI + H_2}$"
-
-    setattr(self, "gas_neutral_H_mass", neutral_H_mass)
+        try:
+            # Fetch dust fields
+            dust_mass_silicates = getattr(
+                catalogue.dust_masses, f"silicates_mass_{aperture_size}_kpc"
+            )
+            dust_mass_graphite = getattr(
+                catalogue.dust_masses, f"graphite_mass_{aperture_size}_kpc"
+            )
+            # All dust mass
+            dust_mass_total = dust_mass_graphite + dust_mass_silicates
+
+        # In case run without dust
+        except AttributeError:
+            dust_mass_total = unyt.unyt_array(np.zeros_like(HI_mass), units="Msun")
 
+        # Compute dust-to-HI mass ratios
+        # Compute gas-mass weighted O over H
+        dust_to_hi_ratio = unyt.unyt_array(np.zeros_like(HI_mass), "dimensionless")
+        # Avoid division by zero
+        mask = HI_mass > 0.0 * HI_mass.units
+        dust_to_hi_ratio[mask] = dust_mass_total[mask] / HI_mass[mask]
+
+        # Add label
+        dust_to_hi_ratio.name = f"M_{{\\rm dust}}/M_{{\\rm HI}} ({aperture_size} kpc)"
+
+        # Register field
+        setattr(self, f"gas_dust_to_hi_ratio_{aperture_size}_kpc", dust_to_hi_ratio)
+
+    return
+
+
+def register_h2_masses(self, catalogue, aperture_sizes):
+
+    # Loop over aperture sizes
     for aperture_size in aperture_sizes:
+
+        # Fetch H2 mass
+        H2_mass = getattr(
+            catalogue.gas_hydrogen_species_masses, f"H2_mass_{aperture_size}_kpc"
+        )
+
+        # Label of the derived field
+        H2_mass.name = f"$M_{{\\rm H_2}}$ ({aperture_size} kpc)"
+
+        # Compute H2 mass with correction due to He
+        He_mass = getattr(catalogue.gas_H_and_He_masses, f"He_mass_{aperture_size}_kpc")
+        H_mass = getattr(catalogue.gas_H_and_He_masses, f"H_mass_{aperture_size}_kpc")
+
+        H2_mass_with_He = H2_mass * (1.0 + He_mass / H_mass)
```

Comment:
@wmfw23 can you please check whether this is the correct way to account for He in the masses of H2?

Reply:
Hey, yes this will do it, basically should amount to a ~ 30% increase in the H2 masses